akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala

```diff
@@ -154,7 +157,8 @@ private[http] final class HttpRequestParser(
       val uriEnd = findUriEnd()
       try {
         uriBytes = input.slice(uriStart, uriEnd)
-        uri = Uri.parseHttpRequestTarget(new ByteStringParserInput(uriBytes), mode = uriParsingMode)
+        uriParser.input = new ByteStringParserInput(uriBytes)
+        uri = uriParser.parseHttpRequestTarget()
```

Comment:
So we can be sure this never happens concurrently because we're inside a stream stage. Cool :+1: